### DISCUSSION TITLE
Result web part and browser language settings (UI langugage property)

### DISCUSSION THREAD

**Discussion by tbergholt** [microsoft-search/pnp-modern-search#3942] [Q&A]
I working on a project to  display only content which is equal to the users display language  and/or browsers language (Data format is en-US) and it's not possible to set the token equal to DetectedLanguage (Data format is en) or SPTranslatedLanguages (Data format is en-us, with small characters).
Have also tried with DetectedLanguage in a refinableString, while DetectedLanguages is not refinable.
The property Language and Search request is set to UI interface Language and the Result web part is displaying content in other language.
The result web part is in format of roll up, while no inbox is connected to the result web part - Do I need to have an Search box before the request is compared with the users display language and/or browsers language (display language is in this project equal to the browser language)?

Shouldn't the settings of Language  and Search request to UI interface language respect the display language and/or the browsers language?

Settings in the SharePoint User Profile is visible with PowerShell as en-US format and the only place where the settings in handling another language is in the User Profile privacy settings where preferred language in equal to danish.

Torben Bergholt

**Comment by kasperbolarsen** [comment]
Hi Torben 
I guess you are using 
{CurrentDisplayLanguage} = en-US
{User.PreferredDisplayLanguage} 1033

Try using either SP Editor or SharePoint Search Query tool to inspect the data returned by the API, and to me it sure looks like SPTranslationLanguage could be usefull ( perhaps combined with  SPIsTranslation in order to check if the page is a translation)
![image](https://github.com/user-attachments/assets/9411e279-4aa3-4455-9a46-299461b9202a)

Unfortunately, the {User.PreferredDisplayLanguage} returns the ISO country code id, so the only solution I know of, had to add another property in the UPA, that is synced with PreferredDisplayLanguage and return a value like de-de

**Reply by tbergholt** [reply]
Hi Kasper
Thanks for responding ... I've already looked at the data with the debug template (and the SharePoint Search Query tool) and hasn't found any data which I can compare with the  format from {CurrentDisplayLanguage}  or from Context (like CultureInfo.CurrentCultureName or CurrentUICultureName). 
So I think the solution, when the web part is not respecting the UI language, is a custom column with the correct format or an if/else change in the template looking at the format (en-US in this example) and filtering out by DetectedLanguage (in the example equal the value 'en'). The  DetectedLanguage is returning the value of the content language (even though the content is in translated format) and can be put into a refinableString.
I actually thought that setting the property of Language and search request to UI internface language was performing a search only content in related to the UI language.

**Comment by tbergholt** [comment]
Discussion closed - Found a solution with using handlebars

**Reply by kasperbolarsen** [reply]
Are you allowed to share some details for the next person with a similar issue ?
